### PR TITLE
simple react hot loading

### PR DESCRIPTION
**Pull request by jeremy-coleman**
just added an alias in the app config and wrapped the app component with the react hot decorator. no babel needed with the react-dom alias

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,8 @@
     "prettify": "yarn prettier \"packages/**/*.{js,ts,tsx}\" --write"
   },
   "devDependencies": {
+     "react-hot-loader":"4.8.4",
+    "@hot-loader/react-dom":"16.8.6",
```
Comment (AviVahl) [inline package.json]: these two should be added using `yarn`:
`yarn add react-hot-loader @hot-loader/react-dom --dev -W`

this way:
- they will be added with a caret (`^`), as this repo respects semver.
- the yarn.lock will be updated.
- they will be sorted alphabetically for readability.

File: packages/app/webpack.config.js [hunk 2]
```diff
@@ -23,13 +29,23 @@ module.exports = {
...
-        extensions: ['.ts', '.tsx', '.mjs', '.js', '.json']
+        extensions: ['.ts', '.tsx', '.mjs', '.js', '.jsx', '.json'],
+        alias:{
+            "react-dom":"@hot-loader/react-dom"
+        }
     },
     plugins: [
+        //new webpack.DefinePlugin({'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV)}),
+        new webpack.HotModuleReplacementPlugin(),
+        new webpack.NamedModulesPlugin(),
```
Comment (AviVahl) [inline packages/app/webpack.config.js]: not needed in webpack@4+ AFAIK:
https://stackoverflow.com/questions/40861768/webpack-plugin-for-module-friendly-names-path-instead-of-a-number

File: packages/app/src/app.tsx [hunk 3]
```diff
@@ -1,8 +1,13 @@
...
 export interface IAppProps {
     text: string;
 }
 
-export const App: React.FunctionComponent<IAppProps> = ({ text }) => <Main text={text} />;
+export const AppBase: React.FunctionComponent<IAppProps> = ({ text }) => {
+    return <Main text={text} />
+};
+
+export let App = hot(AppBase)
```
Comment (AviVahl) [inline packages/app/src/app.tsx]: should be `const`.

File: packages/app/src/app.tsx [hunk 4]
```diff
@@ -1,8 +1,13 @@
 import React from 'react';
 import { Main } from '@sample-monorepo/components';
+import {hot} from 'react-hot-loader/root'
```
Comment (AviVahl) [inline packages/app/src/app.tsx]: run `yarn lint --fix` in root to fix all missing semi-colons automatically. (PR fails linting in CI)

File: packages/app/webpack.config.js [hunk 5]
```diff
@@ -23,13 +29,23 @@ module.exports = {
         ]
     },
     resolve: {
-        extensions: ['.ts', '.tsx', '.mjs', '.js', '.json']
+        extensions: ['.ts', '.tsx', '.mjs', '.js', '.jsx', '.json'],
```
Comment (AviVahl) [inline packages/app/webpack.config.js]: no point in resolving `.jsx` source files if nobody transpiles them :)

File: packages/app/webpack.config.js [hunk 6]
```diff
@@ -23,13 +29,23 @@ module.exports = {
...
+        new webpack.NamedModulesPlugin(),
         new HtmlWebpackPlugin({ title: 'Sample Monorepo App' }),
         new MiniCssExtractPlugin({
             filename: '[name].css',
             chunkFilename: '[name].css'
         })
     ]
 };
+
+//instead of dev/prod configs, i usually do a base, dev, and prod all in 1 file, and do something like config = env === 'prod' ? prod : dev
```
Comment (AviVahl) [inline packages/app/webpack.config.js]: I usually use the function-style config: https://webpack.js.org/configuration/configuration-types#exporting-a-function

Unlike what being shown in their docs, you can detect the current mode more clearly via the second parameter:
```js
module.exports = function (_env, {mode = 'development'}) {
    return {
        mode
    };
};
```
This way, if somebody runs `yarn webpack`, it defaults to development mode.
`yarn webpack --mode production` switches to `production`.

**Comment by jeremy-coleman** [comment]
Will fix shortly. Another thing id like to fix is that the first update does a full replacement and browser reload , but afterwards hot reloading works fine.  Any idea what is causing that?

**Comment by AviVahl** [comment]
It has been a while, so I am closing this.
Feel free to re-open once issues are ironed out and conflicts with current master are resolved.

**Pull request closed without merging**